Update glossary to include more block based terminology

annezazu opened the pull request:
Offering this update to include the terms Block Based Themes, Block Templates, Block Template Parts, and Full Site Editing since these are increasingly becoming necessary terms to refer to in the wider community. Happy to have any of this rephrased as others see fit. cc @kjellr in particular for the block based theme specific terms.

annezazu committed:
Update glossary to include more block based terminology

Offering this update to include the terms Block Based Themes, Block Templates, Block Template Parts, and Full Site Editing since these are increasingly becoming necessary terms to refer to in the wider community. Happy to have any of this rephrased as others see fit.

kjellr commented on docs/designers-developers/glossary.md:
```diff
@@ -19,6 +22,12 @@
+<dd>Templates are HTML files of block markup that map to templates from the standard WordPress template hierarchy, for example index, single or archive. This helps control the front-end defaults of a site that are not edited via the Page Editor or the Post Editor. </dd>
+
+<dt>Block Template Part</dt>
+<dd>Building on Block Templates, these parts help set structure like a Footer or Header that one typically sees in a WordPress site. With Full Site Editing and block based themes, users can create their own arbitrary Template Parts, save those in the database for their site, and re-use them throughout their site.</dd>
```
I'd note that the key here is that template parts are generally _reusable_ parts like a header or footer.

kjellr reviewed (approved):
This looks good, thanks @annezazu! I left one minor comment but other than that it's good to go.

annezazu committed:
Update glossary.md

Merged template and template parts into block template and block template parts
